File: library/src/main/java/org/fbme/lib/iec61499/declarations/AlgorithmDeclaration.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.ContainedElement;
import org.fbme.lib.common.Declaration;
import org.fbme.lib.common.Element;
import org.jetbrains.annotations.Nullable;

public interface AlgorithmDeclaration extends Declaration, ContainedElement {

    @Nullable AlgorithmBody getBody();

    void setBody(@Nullable AlgorithmBody body);

    @Override
    @Nullable BasicFBTypeDeclaration getContainer();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/AlgorithmDeclaration.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.ContainedElement
import org.fbme.lib.common.Declaration

interface AlgorithmDeclaration : Declaration, ContainedElement {
    var body: AlgorithmBody?
    val temporaryVariables: MutableList<ParameterDeclaration>
    override val container: BasicFBTypeDeclaration?
}
